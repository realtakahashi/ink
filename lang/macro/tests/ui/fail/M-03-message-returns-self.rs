use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod missing_message_returns_self {
    #[ink(storage)]
    struct MissingMessageReturnsSelf {}

    impl MissingMessageReturnsSelf {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn missing_self_arg(&self) -> Self {}
    }
}

fn main() {}
