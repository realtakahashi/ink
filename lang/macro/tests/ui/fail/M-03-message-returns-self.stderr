error: ink! messages and methods must not return `Self`
  --> $DIR/M-03-message-returns-self.rs:15:36
   |
15 |         fn missing_self_arg(&self) -> Self {}
   |                                    ^^^^^^^
